qiskit_nature/second_q/algorithms/excited_states_solvers/excited_states_eigensolver.py

```diff
@@ -63,8 +64,8 @@ def solver(self, solver: Union[Eigensolver, EigensolverFactory]) -> None:
     def get_qubit_operators(
```

Comment:
It would not harm to add a message to each place, right, if they are now using the older types that will replaced - in fact arguably we should since for a user it may end up being quite helpful as it should signal more exactly the place(s) they need to change their code. Maybe in conjunction with the setting they end up seeing more deprecation messages - presently there is nothing around setting right, which I seem commented on here elsewhere - but I think we should have them in places like this too.

Reply:
Agreed :+1: So it would be good if as part of this PR we add deprecation warnings to both:
- the newly added setting (commented elsewhere already, indeed)
- all public API places where `PauliSumOp` is an input option